src/js/ui/components/DataTable.js

```diff
@@ -188,6 +188,9 @@ const getSortVal = (value = null, sortType) => {
                 return -Infinity;
             }
             // Drop $ and parseFloat will just keep the numeric part at the beginning of the string
+            if (sortVal.includes('B')) {
```

Comment:
Very thorough to include this :)

PR looks good, I'll merge it some time soon and give better responses to the others, I've been a bit busy lately. Probably this week.